src/lib.rs

```diff
@@ -211,6 +211,59 @@ impl<const N: usize> Bytes<N> {
     }
 }
 
+/// Construct a `Bytes<N>` instance from an array with `N` elements.
+///
+/// Currently, the array is copied, but a more efficient implementation could be used in the
+/// future.
+///
+/// ```
+/// # use heapless_bytes::Bytes;
+/// let bytes: Bytes<3> = Bytes::from([0, 1, 2]);
+/// ```
+///
+/// Length mismatches cause a compiler error:
+/// ```compile_fail
+/// # use heapless_bytes::Bytes;
+/// let bytes: Bytes<3> = Bytes::from([0, 1]);  // does not compile
+/// ```
+/// ```compile_fail
+/// # use heapless_bytes::Bytes;
+/// let bytes: Bytes<3> = Bytes::from([0, 1, 2, 3]);  // does not compile
+/// ```
+impl<const N: usize> From<[u8; N]> for Bytes<N> {
+    fn from(bytes: [u8; N]) -> Self {
+        Self::from(&bytes)
+    }
+}
+
+/// Construct a `Bytes<N>` instance by copying from an array with `N` or less elements.
+///
+/// ```
+/// # use heapless_bytes::Bytes;
+/// let bytes: Bytes<3> = Bytes::from(&[0, 1, 2]);
+/// let shorter_bytes: Bytes<3> = Bytes::from(&[0, 1]);
+/// ```
+///
+/// Overlong input data causes a compiler error:
+/// ```compile_fail
+/// # use heapless_bytes::Bytes;
+/// let bytes: Bytes<3> = Bytes::from(&[0, 1, 2, 3]);  // does not compile
+/// ```
+impl<const N: usize, const M: usize> From<&[u8; M]> for Bytes<N> {
+    fn from(bytes: &[u8; M]) -> Self {
+        struct AssertLessThanEq<const I: usize, const J: usize>;
+        impl<const I: usize, const J: usize> AssertLessThanEq<I, J> {
+            const ASSERT: () = assert!(I <= J);
```

Comment:
I think you can also add a constant message to the `assert` call.

```suggestion
            const ASSERT: () = assert!(I <= J, "Cannot convert infallibly between two arrays when the capacity of the new array is not sufficient");
```